ee/clickhouse/materialized_columns/columns.py

```diff
@@ -0,0 +1,90 @@
+import re
+from datetime import timedelta
+from functools import wraps
+from typing import Dict, no_type_check
+
+from django.utils.timezone import now
+
+from ee.clickhouse.client import sync_execute
+from posthog.settings import CLICKHOUSE_CLUSTER, CLICKHOUSE_DATABASE, TEST
+
+PropertyName = str
+ColumnName = str
+
+
+def cache_for(cache_time: timedelta):
+    def wrapper(fn):
+        @wraps(fn)
+        @no_type_check
+        def memoized_fn(*args, use_cache=not TEST):
+            current_time = now()
+            if (
+                args not in memoized_fn.__cache
+                or current_time - memoized_fn.__cache[args][0] > cache_time
+                or not use_cache
+            ):
+                memoized_fn.__cache[args] = (current_time, fn(*args))
+            return memoized_fn.__cache[args][1]
+
+        memoized_fn.__cache = {}
+        return memoized_fn
+
+    return wrapper
+
+
+@cache_for(timedelta(minutes=15))
+def get_materialized_columns(table: str) -> Dict[PropertyName, ColumnName]:
+    rows = sync_execute(
+        """
+        SELECT comment, name
+        FROM system.columns
```

Comment:
Is there a reason for using the clickhouse system columns over just storing the materialized columns in postgres or something? It seems unnecessary to have to call this query (even cached) every time the filter formatter is used.

Reply:
- Clickhouse has really good introspection capabilties 
- This is a cheap query with minimal overhead.
- Storing it in postgresql would mean duplicating the data 
- Storing it in postgresql would still require fetching it every N minutes

Honestly it seems like the simplest and most robust solution :)

Reply:
sounds good just making sure it was considered